src/kudu/consensus/raft_consensus.cc

```diff
@@ -4048,7 +4048,6 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
                                        rpc::RpcContext* context) {
   MonoDelta wal_wait_timeout = MonoDelta::FromMilliseconds(FLAGS_raft_log_cache_proxy_wait_time_ms);
   MonoTime wal_wait_deadline = MonoTime::Now() + wal_wait_timeout;
-  raft_proxy_num_requests_received_->Increment();
```

Comment:
raft_proxy_num_requests_received_ is initialized in line 380, in function RaftConsensus::Start() right? How does moving to later part of HandleProxyRequest() help with uninitialized value?

Reply:
on line 4060:

RET_RESPOND_ERROR_NOT_OK(CheckRunningUnlocked());

This checks the state of consensus to be `kRunning`. It's only set to `kRunning` on line 518 after everything is initialized.

If it's not kRunning the function bails out immediately. This was the original protection against processing RPC calls before initialization.